src/ChunkedCSV.jl

```diff
@@ -109,19 +110,37 @@ struct ParserSettings
     end
 end
 
-_is_supported_type(::Type{T}) where {T} = false
-_is_supported_type(::Type{T}) where {T<:Union{Int,Float64,String,Char,Bool,DateTime,Date}} = true
-_is_supported_type(::Type{FixedDecimal{T,f}}) where {T<:Union{Int8,Int16,Int32,Int64,Int128,UInt8,UInt16,UInt32,UInt64,UInt128},f} = f <= 8
+_is_supported_type(::Type{T}) where {T} = Parsers.supportedtype(T)
+_is_supported_type(::Type{Nothing}) = true
+_is_supported_type(::Type{GuessDateTime}) = true
+function _is_supported_type(::Type{FixedDecimal{T,f}}) where {T,f}
+    # https://github.com/JuliaMath/FixedPointDecimals.jl/blob/1328b9a372d2285765a7255f154f09ffdd692508/src/FixedPointDecimals.jl#L83-L91
+    n = FixedPointDecimals.max_exp10(T)
+    return f >= 0 && (n < 0 || f <= n)
+end
 function validate_schema(types::Vector{DataType})
-    unsupported_types = filter(!_is_supported_type, types)
+    unsupported_types = unique!(filter(!_is_supported_type, types))
     if !isempty(unsupported_types)
-        err_msg = "Provided schema contains unsupported types: $(join(unique!(unsupported_types), ", "))."
-        any(T->(T <: FixedDecimal), unsupported_types) && (err_msg *= " Note: Currently, only decimals with less than 9 decimal places are supported.")
+        err_msg = "Provided schema contains unsupported types: $(join(unsupported_types, ", "))."
         throw(ArgumentError(err_msg))
     end
     return nothing
 end
 
+# Separate out the types that are not pre-compiled by the parser by default
+# and return them as a single Tuple of unique types which can be passed to
+# _parse_rows_forloop! to trigger recompilation.
+function _custom_types(schema::Vector{DataType})
+    custom_types = sort!(collect(keys(
+            IdDict{Type,Nothing}(
+                T => nothing for T in schema if isnothing(get(Enums._MAPPING, T, nothing))
+            ))),
+        by=objectid
+    )
+    return Tuple{Tuple(custom_types)...}
```

Comment:
why do we convert to a tuple just to splat?
```suggestion
    return Tuple{custom_types...}
```

Reply:
Ah, good catch!